harness/features/allVariables.cloud.test.ts

```diff
@@ -37,6 +38,19 @@ describe('allVariables Tests - Cloud', () => {
                 expect(variablesMap).toMatchObject({})
             })
 
+            it('should throw if variables request fails on user error', async () => {
+                scope
+                    .post(`/client/${client.clientId}/v1/variables`)
+                    .reply(401, { message: 'Invalid sdk token' })
+
+                const response = await client.callAllVariables({
+                    user_id: 'test_user',
+                    email: 'user@gmail.com'
+                }, true)
+                const res = await response.json()
+                expect(res.asyncError).toContain('Invalid sdk token')
```

Comment:
you may want to use the `expectErrorMessageToBeTruthy` helper function in case any SDK doesnt have the exact same error message

Reply:
ah true, i was trying to get around that by using toContain since in theory they should always include the message body somewhere in the exception (at least dotnet and node both do) - do we want to both asserting on that, or just that it throws at all?

Reply:
I think for consistency its fine to assert that it throws at all. If we ever undertake an initiative to ensure that all error messages are the same then we can just update the `expectErrorMessageToBeTruthy` function in one place and be done with it

Reply:
makes sense, fixed!